### PR TITLE
convert ConnectRequestsViewController

### PR DESCRIPTION
**Pull request by billypchan**
## What's new in this PR?

convert ConnectRequestsViewController to Swift

### CODE REVIEW

File: Wire-iOS/Sources/UserInterface/ConnectRequests/ConnectRequestsViewController.swift [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+        
+    }
+    
+    func reload() {
+        tableView.reloadData()
+        
+        if let count = connectionRequests?.count,
+            count != 0 {
+            // Scroll to bottom of inbox
+            tableView.scrollToRow(at: IndexPath(row: count - 1, section: 0), at: .bottom, animated: true)
```
Comment (johnxnguyen) [inline Wire-iOS/Sources/UserInterface/ConnectRequests/ConnectRequestsViewController.swift]: I suggest putting this code in a method and removing the comment.

File: Wire-iOS/Sources/UserInterface/ConnectRequests/ConnectRequestsViewController.swift [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+        configureCell(cell, for: indexPath)
+        return cell
+    }
+    
+    // MARK: - UITableViewDelegate
+    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        if tableView.bounds.size.height <= 0 {
+            return UIScreen.main.bounds.size.height
+        }
+        return tableView.bounds.size.height - 48
```
Comment (johnxnguyen) [inline Wire-iOS/Sources/UserInterface/ConnectRequests/ConnectRequestsViewController.swift]: What's this magic number?

File: Wire-iOS/Sources/UserInterface/ConnectRequests/ConnectRequestsViewController.swift [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+private var ConnectionRequestCellIdentifier = "ConnectionRequestCell"
+
+final class ConnectRequestsViewController: UIViewController, UITableViewDataSource, UITableViewDelegate {
+    var connectionRequests: [ZMConversation]?
```
Comment (johnxnguyen) [inline Wire-iOS/Sources/UserInterface/ConnectRequests/ConnectRequestsViewController.swift]: Why is this optional?

Comment (billypchan) [reply Wire-iOS/Sources/UserInterface/ConnectRequests/ConnectRequestsViewController.swift]: changed to non optional

File: Wire-iOS/Sources/UserInterface/ConnectRequests/ConnectRequestsViewController.swift (hunk 2 above)
Comment (billypchan) [reply Wire-iOS/Sources/UserInterface/ConnectRequests/ConnectRequestsViewController.swift]: not sure but it should be calculate with the height of some UI widgets.

File: Wire-iOS/Sources/UserInterface/ConnectRequests/ConnectRequestsViewController.swift (hunk 1 above)
Comment (billypchan) [reply Wire-iOS/Sources/UserInterface/ConnectRequests/ConnectRequestsViewController.swift]: done

File: Wire-iOS/Sources/UserInterface/ConnectRequests/ConnectRequestsViewController.swift (hunk 2 above)
Comment (billypchan) [reply Wire-iOS/Sources/UserInterface/ConnectRequests/ConnectRequestsViewController.swift]: Checked that this inset for giving a hints to the user that if there is another queued requests. It should be keeped if there are more than 1 request.

File: Wire-iOS/Sources/UserInterface/ConnectRequests/ConnectRequestsViewController.swift [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+    private func configureCell(_ cell: ConnectRequestCell, for indexPath: IndexPath) {
+        /// get the user in reversed order, newer request is shown on top
+        let request = connectionRequests[(connectionRequests.count - 1) - (indexPath.row)]
+        
+        let user = request.connectedUser
+        cell.user = user
+        cell.selectionStyle = .none
+        cell.separatorInset = .zero
+        cell.preservesSuperviewLayoutMargins = false
+        cell.layoutMargins = .zero
```
Comment (typfel) [inline Wire-iOS/Sources/UserInterface/ConnectRequests/ConnectRequestsViewController.swift]: This was previously `UIEdgeInsetsMake(0, 0, 8, 0);`

File: Wire-iOS/Sources/UserInterface/ConnectRequests/ConnectRequestsViewController.swift [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+        
+        cell.acceptBlock = { [weak self] in
+            if self?.connectionRequests.isEmpty == true {
```
Comment (typfel) [inline Wire-iOS/Sources/UserInterface/ConnectRequests/ConnectRequestsViewController.swift]: Could be a guard

File: Wire-iOS/Sources/UserInterface/ConnectRequests/ConnectRequestsViewController.swift [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+                ZClientViewController.shared?.hideIncomingContactRequests() {
+                    if let oneToOneConversation = user?.oneToOneConversation {
+                        ZClientViewController.shared?.select(oneToOneConversation, focusOnView: true, animated: true)
+                    }               }
+            }
+        }
+        
+        cell.ignoreBlock = { [weak self] in
+            if self?.connectionRequests.isEmpty == true {
```
Comment (typfel) [inline Wire-iOS/Sources/UserInterface/ConnectRequests/ConnectRequestsViewController.swift]: Could be a guard

File: Wire-iOS/Sources/UserInterface/ConnectRequests/ConnectRequestsViewController.swift (hunk 2 above)
Comment (johnxnguyen) [reply Wire-iOS/Sources/UserInterface/ConnectRequests/ConnectRequestsViewController.swift]: Magic numbers should be named so we know what they represent.

Comment (johnxnguyen) [reply Wire-iOS/Sources/UserInterface/ConnectRequests/ConnectRequestsViewController.swift]: Ah I see you left a comment there.

**Pull request merged**